satellite/contact/endpoint.go

```diff
@@ -42,23 +47,32 @@ func (endpoint *Endpoint) CheckIn(ctx context.Context, req *pb.CheckInRequest) (
 
 	peerID, err := identity.PeerIdentityFromContext(ctx)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		endpoint.log.Info("failed to get node ID from context", zap.String("node address", req.Address), zap.String("error", err.Error()))
+		return nil, rpcstatus.Error(rpcstatus.Unknown, errCheckInIdentity.New("failed to get ID from context: %v", err).Error())
 	}
 	nodeID := peerID.ID
 
 	err = endpoint.service.peerIDs.Set(ctx, nodeID, peerID)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		endpoint.log.Info("failed to add peer identity entry for ID", zap.String("node address", req.Address), zap.String("node ID", nodeID.String()), zap.String("error", err.Error()))
+		return nil, rpcstatus.Error(rpcstatus.FailedPrecondition, errCheckInIdentity.New("failed to add peer identity entry for ID: %v", err).Error())
 	}
 
 	lastIP, err := overlay.GetNetwork(ctx, req.Address)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		endpoint.log.Info("failed to resolve IP from address", zap.String("node address", req.Address), zap.String("node ID", nodeID.String()), zap.String("error", err.Error()))
+		return nil, rpcstatus.Error(rpcstatus.InvalidArgument, errCheckInNetwork.New("failed to resolve IP from address: %s, err: %v", req.Address, err).Error())
 	}
 
 	pingNodeSuccess, pingErrorMessage, err := endpoint.pingBack(ctx, req, nodeID)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		endpoint.log.Info("failed to ping back address", zap.String("node address", req.Address), zap.String("node ID", nodeID.String()), zap.String("error", err.Error()))
+		err = errCheckInNetwork.New("failed to ping node (ID: %s) at address: %s, err: %v", nodeID, req.Address, err)
```

Comment:
a bit puzzled about this line, line 72, and 73. if we do line 70, 72 will never be true. errCheckInNetwork.New (as opposed to Wrap) is making a brand new error, so errPingBackDial.Has(err) will never be true.

Reply:
oops, let me fix that.